Source/Applications/openXDA/openXDA/Controllers/Widgets/ChannelViewController.cs

```diff
@@ -0,0 +1,68 @@
+//******************************************************************************************************
+//  ChannelViewController.cs - Gbtc
+//
+//  Copyright � 2020, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  10/28/2025 - G. Santos
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+using System;
+using System.Web.Http;
+using GSF.Data;
+using GSF.Data.Model;
+using GSF.Web.Model;
+using openXDA.Model;
+
+namespace openXDA.Controllers.Widgets
+{
+    /// <summary>
+    /// Controller that handles fetching and searching of the <see cref="ChannelDetail"/> model.
+    /// </summary>
+    [RoutePrefix("api/Widgets/ChannelView")]
+    public class ChannelViewController : ModelController<ChannelDetail>
+    {
+        /// <summary>
+        /// Endpoint that mimics ChannelView searching with a <see cref="ParentKeyAttribute"/>,
+        /// but replacing the normal <see cref="Meter"/> marked <see cref="ParentKeyAttribute"/> with <see cref="Event"/> .
```

Comment:
Subtle difference between "replacing the normal Meter marked ParentKeyAttribute with Event." and what is actually happening:
We use the `EventID` to pull a `MeterID` so technically it will include `Channels` that have nothing to do with the event ID supplied - as long as they happen to be associated with that same meter

Reply:
Good point. I have reworded it.

Reply:
That seems a little roundabout. 
So it looks like: 
Widget needs TrendData (and I assume Channel objects?) -> sends EventID to PQB/PQD -> proxies request to here?

That's fine, but in that case are you using any of the other endpoints that come with the `ModelController`? or just this one?
If it's just this one there is no reason to make this a `ModelController` and we probably don't need to get `ParentKey` involved at all. just do a `QueryRecordsWhere("MeterID = (SELECT MeterID FROM EVENT WHERE ID = {0})")`